For expert review

Yurazast opened the pull request.

Yurazast committed:
Uploaded project files

Yurazast committed:
Updated StringListReplaceInStrings and StringListPrint functions

Yurazast committed:
Added menu to interact with string list

IhorShevchuk commented on menu.cpp:
```diff
@@ -0,0 +1,108 @@
+
+void SelectFunction(const int choice)
+{
+	if (choice == 1)
```
This looks like a 'switch' statement not 'if else'. Can we rewrite this to 'switch'?

Yurazast replied:
Yes, we can. But for me if 'switch' does not fit on one page, we need to scroll up to see what variable are using to compare. Also we don't need to forget to place 'break' almost everywhere, so that 'switch' becomes huge.

IhorShevchuk commented on menu.cpp:
```diff
@@ -0,0 +1,108 @@
+
+void ReplaceStringWithStringInList()
+{
+	char before[50], after[50];
```
Can we extract this 50 constant into global constant or define so that it would be easier to manipulate with it in future?

Yurazast replied:
I'll come up with something about it.

IhorShevchuk commented on string_list.cpp:
```diff
@@ -0,0 +1,226 @@
+
+void StringListInit(char*** list)
+{
+	if (list == NULL) return;
```
Error log is missed here.

Yurazast replied:
Ok, I will add it.

Yurazast committed:
Added new function InputStringSafe, updated SelectFunction and other minor improvements

- In SelectFunction if-else was replaced by switch
- Added two defines to be easier to manipulate buffer size and scanf specifier
- Added new function InputStringSafe to get rid of code duplication
- Error log was added for StringListInit function when list is NULL

IhorShevchuk commented on string_list.cpp:
```diff
@@ -0,0 +1,230 @@
+
+	char* new_str = (char*) malloc(sizeof(char) * strlen(str) + 1);
+	if (new_str == NULL)
+		return;
```
Error message is missed here.

IhorShevchuk commented on string_list.cpp:
```diff
@@ -0,0 +1,230 @@
+
+int StringListIndexOf(char** list, String str)
+{
+	if (list == NULL || list[VALUE] == NULL) return -1;
```
Same as above

IhorShevchuk reviewed (commented):
Looks good. Nice work.

Yurazast committed:
Added several error messages
